### PR TITLE
Minor doc fixes

### LINKED ISSUE #4650: Keep the setting of user created DataLoader in replacing DistributedSampler
### Motivation

As mention at #2789,  the default behavior of `replace_sampler_ddp ` is creating a new `DistributedSampler`. The `shuffle` setting depends on the kind of dataloader (train or val/test dataloader). However, this behavior override the setting of user defined dataloader, such as `shuffle` or `drop_last`. A more reasonable solution is to get this setting direly from user created dataloader, and apply the same setting in `DistributedSampler`.

### Pitch

For example, we can get the `shuffle` setting form `dataloader.sampler`. If this is a instance of `SequentialSampler`, `shuffle=False`.

### Alternatives

Set `replace_sampler_ddp=False`, and handle it by hand.

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixed #4650, part of #5099, and some other doc fixes I found while reading.

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/multi_gpu.rst [hunk 1]
```diff
@@ -58,7 +58,7 @@ This will make your code scale to any arbitrary number of GPUs or TPUs with Ligh
         z = torch.Tensor(2, 3)
         z = z.type_as(x)
 
-The :class:`~pytorch_lightning.core.lightning.LightningModule` knows what device it is on. You can access the reference via `self.device`.
+The :class:`~pytorch_lightning.core.lightning.LightningModule` knows what device it is on. You can access the reference via ``self.device``.
 Sometimes it is necessary to store tensors as module attributes. However, if they are not parameters they will
 remain on the CPU even if the module gets moved to a new device. To prevent that and remain device agnostic,
 register the tensor as a buffer in your modules's `__init__` method with :meth:`~torch.nn.Module.register_buffer`.
```
Comment (carmocca) [inline docs/source/multi_gpu.rst:64]:
```suggestion
register the tensor as a buffer in your modules's ``__init__`` method with :meth:`~torch.nn.Module.register_buffer`.
```

**Review by carmocca — APPROVED** [review]

File: docs/source/multi_gpu.rst [hunk 2]
```diff
@@ -99,7 +99,11 @@ Lightning adds the correct samplers when needed, so no need to explicitly add sa
         dataset = MNIST(...)
         return DataLoader(dataset)
 
-.. note:: You can disable this behavior with `Trainer(replace_sampler_ddp=False)`
+.. note::
+    By default it will add ``shuffle=True`` for train sampler and ``shuffle=False`` for val/test sampler.
+    ``drop_last`` in DistributedSampler will be set to its default value in PyTorch.
```
Comment (Borda) [inline docs/source/multi_gpu.rst:104]: let's add ref also for `DistributedSampler`

File: docs/source/trainer.rst [hunk 3]
```diff
@@ -619,7 +619,7 @@ will need to be set up to use remote filepaths.
 
 distributed_backend
 ^^^^^^^^^^^^^^^^^^^
-This has been renamed "accelerator".
+This has been renamed ``accelerator``.
```
Comment (Borda) [inline docs/source/trainer.rst:622]:
```suggestion
Deprecated: This has been renamed ``accelerator``.
```

**Review by Borda — APPROVED** [review]

**Review by ydcjeff — APPROVED** [review]

File: docs/source/multi_gpu.rst [hunk 4]
```diff
@@ -75,8 +75,8 @@ register the tensor as a buffer in your modules's `__init__` method with :meth:`
 
 Remove samplers
 ^^^^^^^^^^^^^^^
-In PyTorch, you must use `torch.nn.DistributedSampler` for multi-node or TPU training. The
-sampler makes sure each GPU sees the appropriate part of your data.
+In PyTorch, you must use `DistributedSampler <https://pytorch.org/docs/stable/data.html#torch.utils.data.distributed.DistributedSampler>`_
```
Comment (awaelchli) [inline docs/source/multi_gpu.rst:78]: You can try this, it should work:
```suggestion
In PyTorch, you must use :class:`~torch.utils.data.distributed.DistributedSampler`
```

File: docs/source/multi_gpu.rst [hunk 5]
```diff
@@ -593,7 +597,7 @@ Below are the possible configurations we support.
 
 Implement Your Own Distributed (DDP) training
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-If you need your own way to init PyTorch DDP you can override :meth:`pytorch_lightning.plugins.ddp_plugin.DDPPlugin.init_ddp_connection`.
+If you need your own way to init PyTorch DDP you can override: :meth:`pytorch_lightning.plugins.ddp_plugin.DDPPlugin.init_ddp_connection`.
```
Comment (awaelchli) [inline docs/source/multi_gpu.rst:600]:
```suggestion
If you need your own way to init PyTorch DDP you can override :meth:`~pytorch_lightning.plugins.ddp_plugin.DDPPlugin.init_ddp_connection`.
```
it's part of the sentence, so the colon should not be necessary? But I suggest to shorten the name with the tilde.

File: docs/source/trainer.rst [hunk 6]
```diff
@@ -1099,7 +1099,9 @@ as you request.
 
 Your effective batch size is batch_size * total tpu cores.
 
-.. note:: No need to add a DistributedDataSampler, Lightning automatically does it for you.
+.. note::
+    No need to add a `DistributedSampler <https://pytorch.org/docs/stable/data.html#torch.utils.data.distributed.DistributedSampler>`_,
```
Comment (awaelchli) [inline docs/source/trainer.rst:1103]:
```suggestion
    No need to add a :class:`~torch.utils.data.distributed.DistributedSampler`,
```

**Review by awaelchli — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Comment by rohitgr7** [comment]
@tchaton @williamFalcon mind review?

**Comment by Borda** [comment]
@tchaton mind review as code-owner? 🐰

**Review by tchaton — APPROVED** [review]
